pytorch_lightning/loops/fit_loop.py

```diff
@@ -324,8 +340,26 @@ def on_run_end(self) -> None:
         self.trainer.strategy.on_train_end()
 
     def teardown(self) -> None:
+        if self._data_fetcher is not None:
+            self._data_fetcher.teardown()
+            self._data_fetcher = None
         self.epoch_loop.teardown()
 
     def _should_accumulate(self) -> bool:
         """Whether the gradients should be accumulated."""
         return self.epoch_loop._should_accumulate()
+
+
+def _select_data_fetcher(trainer: "pl.Trainer") -> Type[AbstractDataFetcher]:
```

Comment:
Why not make this one a static method of the FitLoop?

Reply:
I'd say an independent function is generally preferred (simpler, no self call). But no strong argument